tests/unit/rpc/RPCEngineTests.cpp

```diff
@@ -0,0 +1,513 @@
+};
+
+struct RPCEngineFlowTestCaseBundle {
+    std::string testName;
+    bool isAdmin;
+    std::string method;
+    std::string params;
+    bool forwarded;
+    std::optional<bool> isTooBusy;
+    std::optional<bool> isUnknownCmd;
+    bool handlerReturnError;
+    std::optional<rpc::Status> status;
+    std::optional<boost::json::object> response;
+};
+
+struct RPCEngineFlowParameterTest : public RPCEngineTest, WithParamInterface<RPCEngineFlowTestCaseBundle> {};
+
+static auto
+generateTestValuesForParametersTest()
+{
+    auto const isAdmin = true;
+    auto const isTooBusy = true;
+    auto const neverCalledIsTooBusy = std::nullopt;
+    auto const neverCalledUnknownCmd = std::nullopt;
+    auto const forwarded = true;
+    auto const cmdUnknown = true;
+    auto const handlerReturnError = true;
+
+    return std::vector<RPCEngineFlowTestCaseBundle>{
+        {"ForwardedSubmit",
+         isAdmin,
+         "submit",
+         "{}",
+         forwarded,
+         neverCalledIsTooBusy,
+         neverCalledUnknownCmd,
+         !handlerReturnError,
+         rpc::Status{},
+         boost::json::parse(FORWARD_REPLY).as_object()},
+        {"ForwardAdminCmd",
+         !isAdmin,
+         "ledger",
+         R"JSON({"full": true, "ledger_index": "current"})JSON",
+         !forwarded,
+         neverCalledIsTooBusy,
+         neverCalledUnknownCmd,
+         !handlerReturnError,
+         rpc::Status{RippledError::rpcNO_PERMISSION},
+         std::nullopt},
+        {"BackendTooBusy",
+         !isAdmin,
+         "ledger",
+         "{}",
+         !forwarded,
+         isTooBusy,
+         neverCalledUnknownCmd,
+         !handlerReturnError,
+         rpc::Status{RippledError::rpcTOO_BUSY},
+         std::nullopt},
+        {"HandlerUnknown",
+         !isAdmin,
+         "ledger",
+         "{}",
+         !forwarded,
+         neverCalledIsTooBusy,
+         cmdUnknown,
+         !handlerReturnError,
+         rpc::Status{RippledError::rpcUNKNOWN_COMMAND},
+         std::nullopt},
+        {"HandlerReturnError",
+         !isAdmin,
+         "ledger",
+         R"JSON({"hello": "world", "limit": 50})JSON",
+         !forwarded,
+         neverCalledIsTooBusy,
+         !cmdUnknown,
+         handlerReturnError,
+         rpc::Status{"Very custom error"},
+         std::nullopt},
+        {"HandlerReturnResponse",
+         !isAdmin,
+         "ledger",
+         R"JSON({"hello": "world", "limit": 50})JSON",
+         !forwarded,
+         neverCalledIsTooBusy,
+         !cmdUnknown,
+         !handlerReturnError,
+         std::nullopt,
+         boost::json::parse(R"JSON({"computed": "world_50"})JSON").as_object()},
+    };
+}
+
+INSTANTIATE_TEST_CASE_P(
+    RPCEngineFlow,
+    RPCEngineFlowParameterTest,
+    ValuesIn(generateTestValuesForParametersTest()),
+    tests::util::NameGenerator
+);
+
+TEST_P(RPCEngineFlowParameterTest, Test)
+{
+    auto const testBundle = GetParam();
```

Comment:
nit: I think `GetParam()` returns a reference, so there is no reason to do a copy here.